cql3/query_processor.cc

```diff
@@ -1002,6 +1002,24 @@ query_processor::forward(query::forward_request req, tracing::trace_state_ptr tr
     co_return co_await remote_.get().forwarder.dispatch(std::move(req), std::move(tr_state));
 }
 
+::shared_ptr<messages::result_message> result_with_workaround_for_concurrent_statements(
+        const statements::schema_altering_statement& stmt) {
+    using event_t = cql_transport::event::schema_change;
+    ::shared_ptr<event_t> ce;
+    // in theory we could add the workaround to more (or even all) schema statements
+    // but for now let's limit to what we know causes problems for naive clients
```

Comment:
Everything that can be CREATEd IF NOT EXISTS is prone to the problem described in the issue. I'm not sure why we are limiting ourselves only to keyspaces.

Reply:
>  why we are limiting ourselves only to keyspaces

Following 'if it ain't broke, don't fix it' principle. We can "fix" it for everything but that's bigger scope, are we confident that it's a good fix? Especially that in Cassandra it exists the opposite fix (https://issues.apache.org/jira/browse/CASSANDRA-7600)

Reply:
yes, but be more specific and please refer more to this patch than the others you've made. so you vote for including it for all schema changing statements (alter included) ?

I have no gut feeling what else is broken, so I've included what we know. But I can extend it no problem, if you guys feel it's better.

Reply:
I think all the schema **creation** operations with "IF NOT EXISTS" are vulnerable, for the same reason. You asked about ALTER but I don't think it's relevant - it doesn't have IF NOT EXISTS as far as I know?
Also, DROP is not important to fix - the only "vulnerable" situation is where parallel DROP is followed by a CREATE, which means this is a situation of parallel DROP *and* CREATE, which we don't support pre-raft (and post-raft, is serialized correcly anyway).

The reason I referred in this comment to my patch was specific: I told you that I modified CREATE TABLE, CREATE MATERIALIZED VIEW and CREATE KEYSPACE, and then Kostja thought that this wasn't enough and there are more. So I don't supposed he would now think that just KEYSPACE is enough.